src/promptflow/promptflow/executor/_service/utils/service_utils.py

```diff
@@ -4,25 +4,61 @@
 
 import json
 import os
+from contextlib import contextmanager
 
 from promptflow._core.connection_manager import ConnectionManager
+from promptflow._core.operation_context import OperationContext
 from promptflow._utils.exception_utils import ErrorResponse, ExceptionPresenter, JsonSerializedPromptflowException
-from promptflow._utils.logger_utils import LogContext, logger
+from promptflow._utils.logger_utils import LogContext, service_logger
+from promptflow._version import VERSION
 from promptflow.executor._service.contracts.execution_request import BaseExecutionRequest
 
 
-def get_log_context(request: BaseExecutionRequest):
+@contextmanager
+def noop_context():
+    yield
+
+
+def get_log_context(request):
+    if isinstance(request, BaseExecutionRequest):
+        run_mode = request.get_run_mode()
+        credential_list = ConnectionManager(request.connections).get_secret_list()
+        return LogContext(file_path=request.log_path, run_mode=run_mode, credential_list=credential_list)
+    else:
+        # return a noop context if request is not a execution request
+        return noop_context()
+
+
+def get_service_log_context(request: BaseExecutionRequest):
```

Comment:
Shall we just use `get_log_context`?
Why we need different log context as receiving request and new process?

Reply:
Will update it the next PR.